iso15118/secc/states/iso15118_2_states.py

```diff
@@ -1278,10 +1279,31 @@ async def process_message(
             # [V2G2-847] - The EV shall signal CP State C or D no later than 250ms
             # after sending the first PowerDeliveryReq with ChargeProgress equals
             # "Start" within V2G Communication SessionPowerDeliveryReq.
+
             # [V2G2-860] - If no error is detected, the SECC shall close the Contactor
             # no later than 3s after measuring CP State C or D.
-            # TODO: Before closing the contactor, we may need to check to
+            # Before closing the contactor, we need to check to
             # ensure the CP is in state C or D
+            cp_state = await self.comm_session.evse_controller.get_cp_state()
+            if cp_state not in [CpState.C2, CpState.D2]:
+                logger.info(
+                    f"Cp state is not C2 or D2, state is {cp_state} ."
+                    f"waiting for C2 or D2"
+                )
+                # wait for state C or D 250ms
+                await asyncio.sleep(0.25)
```

Comment:
Just so I understand this correctly - is this 250ms from the specification? Or do we actually have a little less than 3 seconds to detect state C2 or D2?

Reply:
this is from spec req number V2G2-847.  << The EV shall signal CP State C or D no later than 250ms after sending the first PowerDeliveryReq with ChargeProgress equals "Start" within V2G Communication SessionPowerDeliveryReq. >>

Reply:
Done, actually, it was there, I  just rearranged it.